search/agent/base.py

```diff
@@ -0,0 +1,182 @@
+"""
+Provides a base class for Kinesis record handling.
+
+.. _todo: This should move to arXiv-base, per ARXIVNG-281.
```

Comment:
Do TODOs get backlog tickets for tracking?

Reply:
Not automatically. But we should think more about how we want to handle/track debt of this kind